test/runnable/test15.d

```diff
@@ -1390,6 +1390,18 @@ void test19758()
     }
 }
 
+/************************************/
+// https://issues.dlang.org/show_bug.cgi?id=19968
+
+@safe void test19968()
+{
+    int[2] array = [16, 678];
+    union U { int i; bool b; }
+    U u;
+    u.i = 0xDEADBEEF;
+    assert(array[u.b] == 678);
```

Comment:
Shouldn't this throw an out or range error? How is `U.u` == 1 when the memory area was set to dead beef?

Reply:
`0xDEADBEEF & 1` is `1`.

Reply:
Ah of course 👍

Reply:
> `0xDEADBEEF & 1` is `1`.

Nope, `0xDEADBEEF & 1` is `0`.

Reply:
Huh? `0xF` or 15 is not an even number